motr/io_nw_xfer.c

```diff
@@ -873,9 +879,32 @@ static int target_ioreq_iofops_prepare(struct target_ioreq *ti,
 				else
 					buf = bvec->ov_buf[seg];
 
-				rc = m0_rpc_bulk_buf_databuf_add(rbuf,
-					buf, COUNT(ivec, seg),
-					INDEX(ivec, seg), ndom);
+				xfer_len = COUNT(ivec, seg);
+				offset = INDEX(ivec, seg);
+				
+				/*
+				 * Accomodate multiple pages in a single 
+				 * net buffer segment if they are consecutive
+				 * pages
+				 */
+				while (seg < SEG_NR(ivec) && 
+				       xfer_len < max_seg_size) {
+					if (filter == PA_DATA && 
+					    read_in_write && auxbvec != NULL &&
+					    auxbvec->ov_buf[seg] != NULL)
```

Comment:
These checking is similar to Line 874-878.
Can we do something to avoid duplication?